openedx/features/survey_report/api.py

```diff
@@ -0,0 +1,35 @@
+"""
+Contains the logic to manage survey report model.
+"""
+
+from django.conf import settings
+
+from openedx.features.survey_report.models import SurveyReport
+from openedx.features.survey_report.queries import (
+    get_course_enrollments,
+    get_recently_active_users,
+    get_generated_certificates,
+    get_registered_learners,
+    get_unique_courses_offered
+)
+
+
+def generate_report() -> None:
+    """ Generate a report with relevant data."""
+    courses_offered = get_unique_courses_offered()
+    learners = get_recently_active_users(weeks=3)
+    registered = get_registered_learners()
```

Comment:
Not sure if this was discussed before, but registered doesn't say much when you use it in other contexts.
```suggestion
    registered_learners = get_registered_learners()
```

Reply:
So, in this case, perhaps is better "Ever registered learners", right?

Reply:
total_learners ?

Reply:
Finally, I change this to "registered_learners", also, I added the constant MAX_WEEKS_SINCE_LAST_LOGIN. Thanks for your comments @felipemontoya @mariajgrimaldi